DevEdu.API/DevEdu.Business.Tests/Services/UserServiceTests.cs

```diff
@@ -0,0 +1,321 @@
+﻿using DevEdu.Business.Servicies;
+using DevEdu.DAL.Enums;
+using DevEdu.DAL.Models;
+using DevEdu.DAL.Repositories;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace DevEdu.Business.Tests.Services
+{
+    public class UserServiceTests
+    {
+
+        private Mock<IUserRepository> _userRepoMock;
+        private UserDto _userDto;
+        private UserService _userService;
+
+        [SetUp]
+        public void Setup()
+        {
+            _userRepoMock = new Mock<IUserRepository>();
+            _userService = new UserService(_userRepoMock.Object);
+            _userDto = new UserDto
+            {
+                FirstName = "Admin",
+                LastName = "Adminov",
+                Patronymic = "Adminovich",
+                Email = "admin@admin.ad",
+                Username = "Admin01",
+                Password = "qwerty12345",
+                ContractNumber = "admin01",
+                City = (City)1,
+                BirthDate = DateTime.Today,
+                GitHubAccount = "admin/admin.git",
+                Photo = "https://localhost:Admin/admin",
+                PhoneNumber = "adminPhoneNumber",
+                Roles = new List<Role>
+                {
+                     Role.Student ,
+                     Role.Admin
+                }
+            };
+        }
+
+        [Test]
+        public void AddUser_DtoWithRoles_UserDtoWithRolesCreated()
+        {
+            //Given
+            var expectedUserId = 1;
+
+            _userRepoMock.Setup(x => x.AddUser(_userDto))
+                .Returns(expectedUserId)
+                .Verifiable();
+
+            //When
+            var actualUserId = _userService.AddUser(_userDto);
+
```

Comment:
please change the logic of the AddUser method to add roles if they have been provided
and then check created roles as well